printing/printmod.ml

```diff
@@ -440,19 +443,21 @@ let print_module with_body mp =
   let me = Global.lookup_module mp in
   try
     if !short then raise ShortPrinting;
-    unsafe_print_module (Some (Global.env ())) mp with_body me ++ fnl ()
+    unsafe_print_module WithContents
+      (Global.env ()) mp with_body me ++ fnl ()
   with e when CErrors.noncritical e ->
-    unsafe_print_module None mp with_body me ++ fnl ()
+    unsafe_print_module OnlyNames
+      (Global.env ()) mp with_body me ++ fnl ()
 
 let print_modtype kn =
   let mtb = Global.lookup_modtype kn in
   let name = print_kn [] kn in
   hv 1
     (keyword "Module Type" ++ spc () ++ name ++ str " =" ++ spc () ++
-     (try
-	if !short then raise ShortPrinting;
-	print_signature' true (Some (Global.env ())) kn mtb.mod_type
-      with e when CErrors.noncritical e ->
-	print_signature' true None kn mtb.mod_type))
-
-
+     try
+      if !short then raise ShortPrinting;
+      print_signature' true WithContents
+        (Global.env ()) kn mtb.mod_type
+     with e when CErrors.noncritical e ->
+      print_signature' true OnlyNames
+        (Global.env ()) kn mtb.mod_type)
```

Comment:
Umm, this seems like the wrong place to handle this case, shouldn't instead be handled in the proper place the contents are not found?

Reply:
Assuming that I understand your question, this was the choice of @letouzey. If at least one field cannot be printed, then no fields are printed at all. So, it is decided at the toplevel, not locally, and, with that, you get a homogeneous display.

Reply:
Umm, I'd be OK with this choice [tho it seems restrictive], however I am not sure I like the control flow here, with a super exception catch-all.

Indeed, where is the exception this captures being generated?

Reply:
I'm not sure of all exceptions which can be raised when printing in an incorrect state, but at least `Not_found`.

Reply:
The question is "where is it raised". This is why I don't this style of control-flow, because even after looking at the code for a long time I still don't see where the jump comes from.

Reply:
> The question is "where is it raised".

In one or the other of `pr_constr` occurring in the scope of the `try`... We could add a comment...

Reply:
It would be good to really localize this and remove this wacky exception-based control. However given that it seems this PR just preserves the old behavior I don't think it is fair to request the cleanup to be done here.